src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAnAotApp.cs

```diff
@@ -274,6 +309,103 @@ public void NativeAot_hw_runs_with_PackageReference_PublishAot_is_enabled(string
             }
         }
 
+        [RequiresMSBuildVersionTheory("17.0.0.32901")]
+        [InlineData(ToolsetInfo.CurrentTargetFramework)]
+        public void NativeAot_hw_runs_with_PackageReference_PublishAot_is_empty(string targetFramework)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                var projectName = "HellowWorldNativeAotApp";
+                var rid = EnvironmentInfo.GetCompatibleRid(targetFramework);
+
+                var testProject = CreateHelloWorldTestProject(targetFramework, projectName, true);
+
+                // This will add a reference to a package that will also be automatically imported by the SDK
+                testProject.PackageReferences.Add(new TestPackageReference("Microsoft.DotNet.ILCompiler", "7.0.0-*"));
```

Comment:
Did you have a chance to think about how we could remove this source of non-determinism (https://github.com/dotnet/sdk/pull/27098#issuecomment-1212598845)?

I see two options:
* Make it so that we know the version here. Pass it from the build somehow.
* Make it so that we know the version at test build time. AFAIK this gets spilled into a csproj, so the version string could also be `$(TheILCompilerPackageVersion)` and it will expand then at the time the test project is built. We just need something in the SDK to tell us what ILCompiler package it's including.

Reply:
(Maybe this doesn't have to be addressed here, just asking to make sure you saw it)